tutorials/nservicebus-101/lesson-1/tutorial.md

```diff
@@ -0,0 +1,144 @@
+  * If you don't have sufficient privileges to install MSMQ, you can [use the SQL Server transport as a substitute](../using-sql-transport.md). 
+
+
+## Exercise
+
+Let's build something simple to give NServiceBus a try.
+
+
+### Create a solution
+
+First, let's create a basic solution and include the dependencies we need.
+
+ 1. In Visual Studio, create a new project and select the **Console Application** project type.
+ 1. Set the project name to **ClientUI**.
+ 1. Set the Solution Name to **RetailDemo**.
+
+Next, we need to add the NServiceBus NuGet package as a dependency. From the [NuGet Package Manager Console](https://docs.nuget.org/ndocs/tools/package-manager-console), type the following:
+
+```no-highlight
+Install-Package NServiceBus -ProjectName ClientUI
+```
+
+This adds a reference to the NServiceBus.Core assembly to the project. With the proper dependencies in place, we're ready to start writing code.
+
+
+### Configure an endpoint
+
+Although NServiceBus is an async framework, the console application's `Main()` method is not. Therefore, it's necessary to create an entry point into code where we can use the `async`/`await` keywords.
+
+In the **Program.cs** file, modify the code to look like the following:
+
+snippet:EmptyProgram
+
+Now we're ready to create a [**messaging endpoint**](/nservicebus/endpoints/). A messaging endpoint (or just **endpoint**) is a logical component that's capable of sending and receiving messages. An endpoint is hosted within a process, which in this case is a simple console application, but could be a web application or other .NET process.
+
+Let's take a look at all the code required to run an endpoint within a console application as a block, and afterward we'll analyze it line-by-line.
+
+Add this code to your AsyncMain method:
+
+snippet:AsyncMain
+
+Let's go line-by-line and find out exactly what each step is doing.
+
+
+#### Console Title
+
+snippet:ConsoleTitle
+
+When developing NServiceBus systems, we tend to run multiple message endpoints at the same time in separate console windows. If we're not careful, it can become difficult to tell them apart. Giving each one a short, clear name can be helpful to make it easier to find the desired window later on.
+
+In later lessons, we'll expand our project to have several endpoints, and this one will be the test endpoint we'll use to send messages to the rest of the system. Thus we give it the name `ClientUI`.
+
+
+#### EndpointConfiguration
+
+snippet:EndpointName
+
+The `EndpointConfiguration` class is where we define all the settings that determine how our endpoint will operate. The single string parameter `ClientUI` is the [**endpoint name**](/nservicebus/endpoints/specify-endpoint-name.md), which serves as the logical identity for our endpoint, and forms a naming convention by which other things will derive their names, such as the **input queue** where the endpoint will listen for messages to process.
```

Comment:
We is author and reader / teacher and student, fostering a "we're in this together" sort of thing.

Simply calling it an endpoint name, or a logical identity, isn't good enough. It doesn't foster understanding, and that understanding shouldn't be hidden behind a link. We need to give hints at what these things mean (a "for example", without being as exhaustive as the doco link would be) so teh reader can start to build their mental model of what's going on.